From b012ff0f6b284969183863d42a6a14eb77ed0a1d Mon Sep 17 00:00:00 2001
From: Stuart Auchterlonie <stuarta@mythtv.org>
Date: Tue, 6 Jun 2017 22:39:03 +0100
Subject: [PATCH] Refs #13047 - major() and minor() are defined in
 <sys/sysmacros.h>

On BSD platforms they remain defined in <sys/types.h>

(cherry picked from commit eca27c542b70deb58e50ad4581b2d6f6b2e1d3c2)
---
 mythtv/libs/libmythtv/videosource.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mythtv/libs/libmythtv/videosource.cpp b/mythtv/libs/libmythtv/videosource.cpp
index 97618adc80..09dd95073d 100644
--- mythtv/libs/libmythtv/videosource.cpp.old
+++ mythtv/libs/libmythtv/videosource.cpp
@@ -3,7 +3,11 @@
 // Standard UNIX C headers
 #include <unistd.h>
 #include <fcntl.h>
+#if defined(__FreeBSD__) || CONFIG_DARWIN || defined(__OpenBSD__)
 #include <sys/types.h>
+#else
+#include <sys/sysmacros.h>
+#endif
 #include <sys/stat.h>
 
 // C++ headers
